apps/daimo-mobile/src/view/TabNav.tsx

```diff
@@ -73,6 +71,105 @@ export function TabNav() {
   );
 }
 
+export function TabNav() {
+  // Track whether we've onboarded. If not, show OnboardingScreen.
+  const [account] = useAccount();
+  const [isOnboarded, setIsOnboarded] = useState<boolean>(account != null);
+  useEffect(() => {
+    // This is a latch: if we clear the account, go back to onboarding.
+    if (isOnboarded && account == null) setIsOnboarded(false);
+  }, [isOnboarded, account]);
+  // Stay onboarding till it's complete.
+  const onOnboardingComplete = () => setIsOnboarded(true);
+
+  if (!isOnboarded) return <OnboardingScreen {...{ onOnboardingComplete }} />;
+
+  function forModalPresentationIOS({
+    current,
+    next,
+    inverted,
+    layouts: { screen },
+  }: {
+    current: {
+      progress: Animated.AnimatedInterpolation<number>;
+    };
+    next?: {
+      progress: Animated.AnimatedInterpolation<number>;
+    };
+    inverted: Animated.AnimatedInterpolation<-1 | 1>;
+    layouts: {
+      screen: { width: number; height: number };
+    };
+  }) {
+    const progress = add(
+      current.progress.interpolate({
+        inputRange: [0, 1],
+        outputRange: [0, 1],
+        extrapolate: "clamp",
+      }),
+      next
+        ? next.progress.interpolate({
+            inputRange: [0, 1],
+            outputRange: [0, 1],
+            extrapolate: "clamp",
+          })
+        : 0
+    );
+
+    const translateY = multiply(
+      progress.interpolate({
+        inputRange: [0, 1, 2],
+        outputRange: [screen.height, 0, 0],
+      }),
+      inverted
+    );
+
+    const overlayOpacity = progress.interpolate({
+      inputRange: [0, 1, 1.0001, 2],
+      outputRange: [0, 0.3, 1, 1],
+    });
+
+    return {
+      cardStyle: {
+        transform: [{ translateY }],
+      },
+      overlayStyle: { opacity: overlayOpacity },
+    };
+  }
+
+  return (
+    <MainStack.Navigator initialRouteName="MainTabNav">
+      <MainStack.Group>
+        <MainStack.Screen
+          name="MainTabNav"
+          component={TabNavigator}
+          options={{ headerShown: false }}
+        />
+      </MainStack.Group>
+      <MainStack.Group
+        screenOptions={{
+          presentation: "modal",
+          headerShown: false,
+          cardStyle: {
+            backgroundColor: "transparent",
+          },
+        }}
+      >
+        <MainStack.Screen
+          name="LinkErrorModal"
+          component={ErrorScreen}
+          options={{
+            ...TransitionPresets.ModalPresentationIOS,
+            detachPreviousScreen: false,
+            gestureResponseDistance: WINDOW_HEIGHT,
+            cardStyleInterpolator: forModalPresentationIOS,
```

Comment:
looks reasonable to me

i'm curious why this is implemented differently from the shake-gesture Send Debug Log (which is also a bottom-sheet popup that can appear on any screen at any time)

Reply:
When I created Send Debug Log there was no stack navigator installed and it could be opened just from one source: The shake gesture. I think it was reasonable not to rewrite some part of navigation to add it even thou it can be opened just from one listener. Thats why I decided to use Bottom Sheet that was already in the app and use it for that purpose.

This bad deeplink can be called from anywhere in the app (QRScreen, Home Screen and maybe some more places in the future) so I kinda wanted to have it in the navigation to make it reasonable to use from anywhere. And for that I needed to heavily modify the navigation and install stack navigator.

Now with navigation rewritten to support modals like this I think this is a great idea to move Send Debug Log modal to work exactly the same as this new Bad Deeplink modal.

Maybe create ticket for it and make it as follow up PR?

Reply:
yeah was just a question! looks good to me